transport/http2_server.go

```diff
@@ -285,19 +300,17 @@ func newHTTP2Server(conn net.Conn, config *ServerConfig) (_ ServerTransport, err
 // operateHeader takes action on the decoded headers.
 func (t *http2Server) operateHeaders(frame *http2.MetaHeadersFrame, handle func(*Stream), traceCtx func(context.Context, string) context.Context) (close bool) {
 	streamID := frame.Header().StreamID
-	var state decodeState
-	for _, hf := range frame.Fields {
-		if err := state.processHeaderField(hf); err != nil {
-			if se, ok := err.(StreamError); ok {
-				t.controlBuf.put(&cleanupStream{
-					streamID: streamID,
-					rst:      true,
-					rstCode:  statusCodeConvTab[se.Code],
-					onWrite:  func() {},
-				})
-			}
-			return
+	state := decodeState{serverSide: true}
+	if err := state.decodeResponseHeader(frame); err != nil {
```

Comment:
`decodeResponseHeader` is supposed to be used on the client side. Perhaps, rename the function to something more appropriate since it's being used on the server as well.

Reply:
Alternatively, add the `if frame.Truncated` check right at the beginning of this function itself.

Reply:
Changed to `decodeHeader`.
Tried to move `if frame.Truncated`, but it seems to only save an declaration for a decodeState variable, but complicates the code substantially, e.g. add redundant code for error handling, have to specialize for server and client.